lib/northbound_grpc.cpp

```diff
@@ -278,18 +272,16 @@ class NorthboundImpl
 				return;
 			}
 
-			mypaths->pop_back();
-
-			tag->async_responder.Write(response, tag);
+			tag->async_responder.WriteAndFinish(
+				response, grpc::WriteOptions(),
+				grpc::Status::OK, tag);
+			tag->state = FINISH;
```

Comment:
So we always finish because the definition of a "complete" return value is now a single chunk, correct?

Reply:
Correct. Long term I think we should have two separate RPCs to fetch state data:
* `Get(xpath, input_offset, chunk_size)` which will be an unary RPC that returns a single chunk of data;
* `GetStream(xpath, chunk_size)` which will return a stream of data chunks (generated asynchronously to avoid blocking the main pthread for too long).

For now `Get()` is a streaming RPC but in practice it only returns one response per request. This is something I couldn't address on this PR due to time constraints.